fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/GroupByQuery.java

```diff
@@ -145,6 +146,81 @@ public GroupByQuery(
         );
     }
 
+    /**
+     * Constructor.
+     *
+     * @param dataSource  The datasource
+     * @param granularity  The granularity
+     * @param dimensions  The dimensions
+     * @param filter  The filter
+     * @param having  The having clause
+     * @param aggregations  The aggregations
+     * @param postAggregations  The post-aggregations
+     * @param intervals  The intervals
+     * @param limitSpec  The limit specification
+     * @param context  The context
+     * @param incrementQueryId  true to fork a new context and bump up the query id, or false to create an exact copy
+     * of the context.
+     * @param virtualColumn The virtual columns
+     */
+    protected GroupByQuery(
+            DataSource dataSource,
+            Granularity granularity,
+            Collection<Dimension> dimensions,
+            Filter filter,
+            Having having,
+            Collection<Aggregation> aggregations,
+            Collection<PostAggregation> postAggregations,
+            Collection<Interval> intervals,
+            LimitSpec limitSpec,
+            QueryContext context,
+            boolean incrementQueryId,
+            Collection<VirtualColumn> virtualColumn
+    ) {
+        super(
```

Comment:
Same advice as above, this becomes the primary constructor and non test invocations should migrate to it.  It's ugly, but it minimizes the changes of throwing virtual columns away under copy-with-modification which we do all over the place.